llvm/lib/Target/AMDGPU/SIProgramInfo.cpp

```diff
@@ -199,3 +201,28 @@ const MCExpr *SIProgramInfo::getPGMRSrc2(CallingConv::ID CC,
 
   return MCConstantExpr::create(0, Ctx);
 }
+
+uint64_t SIProgramInfo::getFunctionCodeSize(const MachineFunction &MF) {
```

Comment:
This function isn't even correct. We really should fix this to emit the MC resolved code size. Can we do that now?

Reply:
I wanted to look at this separately. Right now the problem is AsmPrinter emits end function label into an incorrect place, actually into a kernel descriptor in .rodata. This is even a wrong section. That will take more and really a separate thing, but when fixed I could replace that with MCExpr. I.e., I can emit a separate end label, but this is also a hack.

Reply:
https://github.com/llvm/llvm-project/pull/127549

Reply:
Hm... I have the draft, but the output comes out as
```
.amdhsa_inst_pref_size ((((255*((((.Lfunc_end0-large)+127)/128)>255))+((((.Lfunc_end0-large)+127)/128)*((((.Lfunc_end0-large)+127)/128)<=255)))<<4)&4080)>>4
```
I am not sure we really want this.